sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -268,7 +339,7 @@ class CudfSum(ref: Expression) extends CudfAggregate(ref) {
   //
   @transient val rapidsSumType: DType = GpuColumnVector.getNonNestedRapidsType(dataType)
 
-  override val updateReductionAggregateInternal: cudf.ColumnVector => cudf.Scalar = {
+  override val reductionAggregate: cudf.ColumnVector => cudf.Scalar = {
     dataType match {
       case _: DecimalType =>
         (col: cudf.ColumnVector) => withResource(col.castTo(rapidsSumType)) { tmp =>
```

Comment:
I would have expected the `GpuAggregateFunction` invoking `CudfSum` to cast in `inputProjection` and `preMerge`. Do we need to to cast inside `CudfSum`?